### PR TITLE
changes done as per code review

### PR DESCRIPTION
**Pull request by hkambham**
added all chnages as per request

### CODE REVIEW

File: src/textForCLIAndError.ts [hunk 1]
```diff
@@ -46,6 +47,9 @@ export enum ErrorMessageExt {
 }
 
 export enum textForInfoMsg{
+    downloadFlameExtManually = "You can manually download the extension 'Flame Chart Visualizer for JavaScript Profiles' at URL: https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-js-profile-flame (or) Goto  Extensions(shift+cmd+X) in vscode terminal --> search for 'Flame Chart Visualizer for JavaScript Profiles' --> click on install",
+    cpuprofileFileDownloadSuccess = 'Successfully downloaded the extension Flame Chart Visualizer for JavaScript Profiles.',
+    cpuProfileOptionMsg = `To view '.cpuprofile' files you should have extension 'Flame Chart Visualizer for JavaScript Profiles'. Click on 'Download' to automatically download the extension.`,
```
Comment (bmatthew) [inline src/textForCLIAndError.ts:52]:
```suggestion
    cpuProfileOptionMsg = `To view '.cpuprofile' files you should have extension 'Flame Chart Visualizer for JavaScript Profiles.' Click on 'Download' to automatically download the extension.`,
```

File: package.json [hunk 2]
```diff
@@ -221,49 +222,62 @@
...
+		"esbuild": "^0.15.9",
+		"eslint-import-resolver-node": "^0.3.6",
+		"mocha": "^10.0.0",
+		"ts-loader": "^9.4.0",
 		"typescript": "^4.3.2",
-		"vscode": "^1.1.37",
 		"vscode-extension-tester": "^4.1.2",
-		"vscode-test": "^1.5.2"
+		"vscode-test": "^1.5.2",
+		"webpack-cli": "^4.10.0"
```
Comment (bmatthew) [inline package.json:247]:
```suggestion
		"webpack": "^5.74.0",
		"webpack-cli": "^4.10.0"
```

File: src/extension.ts [hunk 3]
```diff
@@ -64,16 +62,17 @@ export const activate = async function(context: vscode.ExtensionContext){
...
-        .then(async ()=> {     
+        .then(async ()=> {    
         await vscode.window.withProgress({
                 location: vscode.ProgressLocation.Notification,
                 title: "Refreshing EdgeWorker Details",
                 cancellable: true
             }, async (progress, token) => {
                 token.onCancellationRequested(() => {
-                    console.log("User canceled the long running operation");
+                    throw "Fetching edgeworkers cancelled";
```
Comment (bmatthew) [inline src/extension.ts:72]:
```suggestion
                    throw "Fetching EdgeWorkers cancelled";
```

File: src/codeProfilerFunction.ts [hunk 4]
```diff
@@ -2,22 +2,26 @@
...
+import * as fs from 'fs';
+import * as path from 'path';
+import * as dns from 'dns';
 import axios from 'axios';
 import { Agent } from 'https';
-const util = require('util');
-const stream = require('stream');
 const extract = require('extract-json-from-string');
-const pipeline = util.promisify(stream.pipeline);
+const homedir = require('os').homedir();
```
Comment (bmatthew) [inline src/codeProfilerFunction.ts:12]:
```suggestion
const tmpdir = require('os').tmpdir();
```
And then adjusting homedir below

File: src/textForCLIAndError.ts (hunk 1 above)
Comment (hkambham) [reply src/textForCLIAndError.ts:52]: I think this sugesstion is by mistake

File: package.json (hunk 2 above)
Comment (hkambham) [reply package.json:247]: done

File: src/extension.ts (hunk 3 above)
Comment (hkambham) [reply src/extension.ts:72]: done

File: src/textForCLIAndError.ts (hunk 1 above)
Comment (bmatthew) [reply src/textForCLIAndError.ts:52]: It's a minor thing but American style quoting puts punctuation inside quote marks even when they're not part of a quoted text, and American style text is the default for UI.

So either way, I'm not overly picky about this.

**Review by bmatthew — APPROVED** [review]

File: src/textForCLIAndError.ts (hunk 1 above)
Comment (hkambham) [reply src/textForCLIAndError.ts]: hoo I did not knew this.. will change NP

File: src/codeProfilerFunction.ts (hunk 4 above)
Comment (hkambham) [reply src/codeProfilerFunction.ts]: done

**Comment by hkambham** [comment]
done code changes

**Pull request merged**